index.d.ts

```diff
@@ -0,0 +1,456 @@
+declare module 'oidc-provider' {
+    import { Request, Response, Handler } from 'express';
+    import Koa = require('koa');
+    import Router = require('koa-router');
+    import { IncomingMessage, ServerResponse } from 'http';
+    import EventEmitter = require('events');
+    import { JWK } from 'node-jose';
```

Comment:
iirc the types for node-jose are in an awful state, what would it mean if we were not to import them?

Reply:
In my opinion, the less we import the better as this might require us to state the additional types required in our `package.json` `devDependencies`.